tests/prague/eip7702_set_code_tx/test_set_code_txs.py

```diff
@@ -378,6 +378,81 @@ def test_set_code_to_tstore_reentry(
     )
 
 
+@pytest.mark.parametrize(
+    "call_opcode",
+    [
+        Op.CALL,
+        Op.STATICCALL,
+    ],
+)
```

Comment:
```suggestion
@pytest.mark.with_all_call_opcodes(selector=lambda call_opcode: call_opcode not in [Op.DELEGATECALL, Op.CALLCODE, Op.EXTDELEGATECALL])
```

Should be equivalent and should automatically fill the test with EOF call opcodes when they are enabled.

Reply:
Sorry this param should be removed, in this context anything besides CALL makes no sense, since STATICCALL cannot SSTORE and all other opcodes will run in the context (i.e. where to write any store to) of the call chain contract address, which is not the purpose of this test (will remove this `call_opcode` thing).

Nevertheless, a good note to know for the future to select "all but not" :smile:

Reply:
I have added this, it now should select CALL and EXTCALL. Should I also add rules for when it runs in EOF mode? Because I think it now will deploy legacy code when it expects an EOF container? (For EXTCALL?)

Reply:
> Nevertheless, a good note to know for the future to select "all but not" 😄

Yes and the reasoning behind this (instead of directly selecting the opcodes) is that the test might fail on newer forks where another `*CALL*` opcode gets added, but this is a good thing because it prompts to update the tests.

Reply:
> I have added this, it now should select CALL and EXTCALL. Should I also add rules for when it runs in EOF mode? Because I think it now will deploy legacy code when it expects an EOF container? (For EXTCALL?)

I just tried it with the `PragueEIP7692` fork and filling with besu (supports 7702+EOF) and it worked flawlessly as is right now, so no changes needed :100:

Reply:
Just to elaborate a bit further this is because `pre.deploy_contract` automatically detects (1) the code being non-terminating `Bytecode` and appending the `Op.STOP`, and (2) the `evm_code_type` fixture automatically modifying the behavior of the contract deployment.